### PR TITLE
Ignite 2190

### PR DESCRIPTION
**Pull request by nizhikov**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanWithEventsSelfTest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.apache.ignite.internal.processors.cache;
```
Comment (daradurvs) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanWithEventsSelfTest.java:18]: Missed Apache license.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanSelfTest.java [hunk 2]
```diff
@@ -90,6 +90,13 @@
     }
 
     /**
+     * @return EventTypes to record
```
Comment (daradurvs) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanSelfTest.java]: The dot in the end.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java [hunk 3]
```diff
@@ -537,6 +537,18 @@ public boolean isRecordable(int type) {
     }
 
     /**
+     * Checks whether this event type has any listener
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java]: The dot in the end.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java [hunk 4]
```diff
@@ -537,6 +537,18 @@ public boolean isRecordable(int type) {
...
+     *
+     * @param type Event type to check.
+     * @return Whether or not this event type has any listener
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java]: The dot in the end.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java [hunk 5]
```diff
@@ -1371,6 +1372,18 @@ protected void runQuery(GridCacheQueryInfo qryInfo) {
...
+                    U.quietAndInfo(log, "Suggestion for the cause of ClassNotFoundException");
+                    U.quietAndInfo(log, "To disable, set -D" + IGNITE_QUIET + "=true");
+                    U.quietAndInfo(log, "  ^-- Ignite configured to use BinaryMarshaller but keepBinary is false for " +
+                        "request");
+                    U.quietAndInfo(log, "  ^-- Server node need to load definition of data classes. " +
+                        "It can be reason of ClassNotFoundException(consider IgniteCache.withKeepBinary to fix)");
+                    U.quietAndInfo(log, "Refer this page for detailed information: " +
+                        "https://apacheignite.readme.io/docs/binary-marshaller");
+                }
+
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java]: Please, investigate an posibility to use GridPerformanceSuggestions class for this block.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanWithEventsSelfTest.java (hunk 1 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanWithEventsSelfTest.java:18]: fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanSelfTest.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheBinaryObjectsScanSelfTest.java]: done

File: modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java (hunk 3 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java]: done

File: modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java (hunk 4 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/eventstorage/GridEventStorageManager.java]: done

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java (hunk 5 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java]: I'm not sure if it *performance* suggestion.
Moreover performance suggestion logged on node start and this suggestion logged only for certain exception(ClassNotFoundException) with certain configuration(BinaryMarhsaller and keepBinary=false)

Comment (daradurvs) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java]: Got it.

**Pull request closed without merging**